packages/ui/src/types/authenticator/form.ts

```diff
@@ -0,0 +1,35 @@
+export type AuthFormData = Record<string, string>;
+
+export type formFieldComponents =
+  | 'signIn'
+  | 'signUp'
+  | 'forceNewPassword'
+  | 'confirmResetPassword'
+  | 'confirmSignIn'
+  | 'confirmSignUp'
+  | 'confirmVerifyUser'
+  | 'resetPassword'
+  | 'setupTOTP';
+
+export type FormFields = {
+  [key in formFieldComponents]?: formField;
+};
+export interface formField {
+  [key: string]: formFieldTypes;
+}
+
+export interface formFieldTypes {
+  labelHidden?: boolean;
```

Comment:
I know this is unrelated to your refactor, but shouldn't these have JSDoc comments for intellisense?

Reply:
Yep, we should especially for public ones. Let me add some while I'm add it.

Reply:
Done! https://github.com/aws-amplify/amplify-ui/pull/1456/commits/977acc65575398ba1c7d337f66f2023b11a1102b